update.go

```diff
@@ -11,31 +11,49 @@ import (
 	"fmt"
 	"io"
 	"net/http"
-	"os"
 
 	"github.com/edgexfoundry/edgex-go/pkg/models"
 )
 
 func callbackHandler(w http.ResponseWriter, req *http.Request) {
-	// use req.Method vs. method
-
+	defer req.Body.Close()
 	dec := json.NewDecoder(req.Body)
 	cbAlert := models.CallbackAlert{}
 
 	err := dec.Decode(&cbAlert)
 	if err != nil {
-		// TODO: handle error properly
-		fmt.Fprintf(os.Stderr, "service: callbackHandler invalid request: %v\n", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		svc.lc.Error(fmt.Sprintf("Invalid callback request: %v", err))
+		return
 	}
 
-	action := cbAlert.ActionType
-	id := cbAlert.Id
+	if (cbAlert.Id == "") || (cbAlert.ActionType == "") {
```

Comment:
is this covers cases when body has follow structure:
1. {}
2. {Id: null}

?

Reply:
Could you please add unit tests for it?

Reply:
Yes, covers those cases

Reply:
tests?